File: csharp/src/AutomationFramework.Core/Models/ProductImage.cs
using System.Text.Json.Serialization;

namespace AutomationFramework.Core.Models;

/// <summary>Image metadata attached to a product, including its attribution.</summary>
public sealed record ProductImage(
    [property: JsonPropertyName("id")] string Id,
    [property: JsonPropertyName("file_name")] string FileName,
    [property: JsonPropertyName("title")] string? Title,
    [property: JsonPropertyName("by_name")] string? ByName,
    [property: JsonPropertyName("by_url")] string? ByUrl,
    [property: JsonPropertyName("source_name")] string? SourceName,
    [property: JsonPropertyName("source_url")] string? SourceUrl);

File: java/src/main/java/com/company/automation/models/ProductImage.java
package com.company.automation.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

/** Image metadata attached to a product, including its attribution. */
@JsonIgnoreProperties(ignoreUnknown = true)
public record ProductImage(
    @JsonProperty("id") String id,
    @JsonProperty("file_name") String fileName,
    @JsonProperty("title") String title,
    @JsonProperty("by_name") String byName,
    @JsonProperty("by_url") String byUrl,
    @JsonProperty("source_name") String sourceName,
    @JsonProperty("source_url") String sourceUrl) {}
